OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/preferencecenter/OptimovePreferenceCenter.java

```diff
@@ -0,0 +1,294 @@
+        UserInfo userInfo = Optimove.getInstance().getUserInfo();
+
+        if (userInfo.getUserId() == null || Objects.equals(userInfo.getUserId(), userInfo.getVisitorId())) {
+            Log.d(TAG, "Customer ID is not set");
+            return;
+        }
+
+        Runnable task = new SetPreferencesRunnable(preferencesSetHandler, updates);
+        executorService.submit(task);
+    }
+
+    static class GetPreferencesRunnable implements Runnable {
+        private final PreferencesGetHandler callback;
+
+        GetPreferencesRunnable(PreferencesGetHandler callback) {
+            this.callback = callback;
+        }
+
+        @Override
+        public void run() {
+            String mappedRegion = shared.mapRegion(config.getRegion());
+            HttpClient httpClient = HttpClient.getInstance();
+            Preferences preferences = null;
+            try {
+                String encodedCustomerId = URLEncoder.encode(Optimove.getInstance().getUserInfo().getUserId(), "UTF-8");
+                String url = "https://preference-center-" + mappedRegion + ".optimove.net/api/v1/preferences?customerId=" + encodedCustomerId + "&brandGroupId=" + config.getBrandGroupId();
+                Response response = httpClient.getSync(url, Optimove.getInstance().getTenantInfo().getTenantId());
+
+                if (!response.isSuccessful()) {
+                    logFailedResponse(response);
+                } else {
+                   preferences = mapResponseToPreferences(response);
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
+            this.fireCallback(preferences);
+        }
+
+        private void fireCallback(@Nullable Preferences preferences) {
+            handler.post(() -> GetPreferencesRunnable.this.callback.run(preferences));
+        }
+    }
+
+    static class SetPreferencesRunnable implements Runnable {
+        private final PreferencesSetHandler callback;
+        private final List<PreferenceUpdate> updates;
+
+        SetPreferencesRunnable(PreferencesSetHandler callback, List<PreferenceUpdate> updates) {
+            this.callback = callback;
+            this.updates = updates;
+        }
+
+        @Override
+        public void run() {
+            String mappedRegion = shared.mapRegion(config.getRegion());
+            HttpClient httpClient = HttpClient.getInstance();
+            boolean result = false;
+            try {
+                String encodedCustomerId = URLEncoder.encode(Optimove.getInstance().getUserInfo().getUserId(), "UTF-8");
+                String url = "https://preference-center-" + mappedRegion + ".optimove.net/api/v1/preferences?customerId=" + encodedCustomerId + "&brandGroupId=" + config.getBrandGroupId();
+                JSONArray data = mapPreferenceUpdatesToArray(updates);
+                Response response = httpClient.putSync(url, data, Optimove.getInstance().getTenantInfo().getTenantId());
```

Comment:
worth using try with resources here? as i understand OKHttp still closes resources if they become idle, but `try with resources` does it faster

just wrap into another try-catch